torch_geometric/data/collate.py

```diff
@@ -124,12 +124,16 @@ def _collate(
     if isinstance(elem, Tensor):
         # Concatenate a list of `torch.Tensor` along the `cat_dim`.
         # NOTE: We need to take care of incrementing elements appropriately.
-        cat_dim = data_list[0].__cat_dim__(key, elem, stores[0])
+        if isinstance(key, str):
```

Comment:
```suggestion
        key = str(key)
```
Can‘t we move this to a one-liner?

Reply:
As long as this doesn't mutate key from the outer frame this should be fine. If it does mutate it, then this may have side effects (such as making all the keys in any collated dictionaries strings instead of other hashable types, ie ints)

Reply:
I think this should be fine :)